src/inmanta/moduletool.py

```diff
@@ -901,23 +910,71 @@ def __init__(self, module_path: str) -> None:
         """
         self._module = ModuleV2(project=None, path=os.path.abspath(module_path))
 
-    def build(self, output_directory: str) -> str:
+    def build(self, output_directory: str, byte_code: bool = False) -> str:
         """
         Build the module and return the path to the build artifact.
+
+        :param byte_code: When set to true, only bytecode will be included. This also results in a binary wheel
         """
         if os.path.exists(output_directory):
             if not os.path.isdir(output_directory):
                 raise ModuleBuildFailedError(msg=f"Given output directory is not a directory: {output_directory}")
+
         with tempfile.TemporaryDirectory() as tmpdir:
             # Copy module to temporary directory to perform the build
             build_path = os.path.join(tmpdir, "module")
             shutil.copytree(self._module.path, build_path)
             self._ensure_plugins(build_path)
             self._move_data_files_into_namespace_package_dir(build_path)
+            if byte_code:
+                self._byte_compile_code(build_path)
+
             path_to_wheel = self._build_v2_module(build_path, output_directory)
             self._verify_wheel(build_path, path_to_wheel)
             return path_to_wheel
 
+    def _byte_compile_code(self, build_path: str) -> None:
+        # Backup src dir and replace .py files with .pyc files
+        for root, dirs, files in os.walk(os.path.join(build_path, "inmanta_plugins"), followlinks=True):
+            for current_file in [f for f in files if f.endswith(".py")]:
+                py_file = os.path.join(root, current_file)
+                pyc_file = f"{py_file}c"
+                py_compile.compile(file=py_file, cfile=pyc_file, doraise=True)
+                os.remove(py_file)
+
+        # Make sure there is a pyc line in the manifest.in
+        with open(os.path.join(build_path, "MANIFEST.in"), "r+", encoding="utf-8") as fh:
+            content = fh.read()
+            # Make sure that there is at least one .pyc include
+            if ".pyc" not in content:
```

Comment:
This could also be "global-exclude *.pyc`, but I don't see how we can reliably handle all such edge cases without overcomplicating this.

Reply:
Unless we just write it regardless? This is only the builder so nothing we write is actually persisted.

Reply:
see also the previous comment, then they know what they are doing :-)